content/blog/2-baby-steps-into-vr.md

```diff
@@ -0,0 +1,359 @@
+public class PlayerMotorDrag: MonoBehaviour
+{
+  /*
+   * When trigger is down moves in the direction facing the main camera.
+   * Accelerate and decelerate progressively with max speed capped by drag.
+   */
+
+    [SerializeField]
+    private float acceleration = 8f;
+    [SerializeField]
+    [Tooltip("Increase to reduce max speed.")]
+    private float movementDrag = 1f;
+    [SerializeField]
+    [Tooltip("Increase to increase brake speed.")]
+    private float decelerationDrag = 1.5f;
+
+    private Rigidbody rb;
+    private bool triggerDown;
+
+    void Awake()
+    {
+        rb = GetComponent<Rigidbody>();
+        rb.useGravity = false;
+        rb.freezeRotation = true;
+        triggerDown = false;
+    }
+
+    void FixedUpdate()
+    {
+        if (Input.GetButtonDown("Fire1"))
+        {
+            rb.drag = movementDrag;
+            triggerDown = true;
+        }
+
+        if (Input.GetButtonUp("Fire1"))
+        {
+            triggerDown = false;
+            rb.drag = decelerationDrag;
+        }
+
+        if (triggerDown)
+        {
+            rb.AddForce(Camera.main.transform.forward * acceleration);
+        }
+    }
+}
+```
+
+The key parts of this code are:
+
+* [`rb.useGravity = false;`](https://docs.unity3d.com/ScriptReference/Rigidbody-useGravity.html "Rigidbody.useGravity") => Disable gravity for the player.
+* `rb.drag = movementDrag;` => Set the drag value when moving, which limit the max speed.
+* `rb.drag = decelerationDrag;` => Set the drag value when not moving, which modify the brake speed.
+* [`rb.AddForce(Camera.main.transform.forward * acceleration);`](https://docs.unity3d.com/ScriptReference/Rigidbody.AddForce.html "Rigidbody.AddForce") => move in the direction you are facing by applying a force to the *Rigidbody*.
+
+
+
+Drag and drop this script on the *Player* and remove the old *PlayerMotorRigidbody* from it (right click on the component and `Remove Component`). Now click the *Play* button and I believe you can fly (and touch the sky)!
+
+
+### Nav Mesh Agent
+This one uses a component called [*Nav Mesh Agent*](https://docs.unity3d.com/Manual/class-NavMeshAgent.html "NavMeshAgent") that can use [*paths*](https://docs.unity3d.com/Manual/Navigation.html "Navigation and Pathfinding") to move the player around. When you press and release the trigger the *Player* will move automatically to the point you were looking at. It will take the shortest path **and avoid obstacle**, which means that the trajectory will not be a straight line if there are obstacles on the way. The advantage is that the user can look around when moving toward his goal, the disadvantage is that he doesn't have direct control over the trajectory (because of the obstacle avoidance) which can cause [**cybersickness**](https://en.wikipedia.org/wiki/Virtual_reality_sickness "Virtual reality sickness") (I will cover cybersickness, how it can occur and how it can be prevented/reduced, in an other section).
+
```

Comment:
path **and avoid obstacle(S)**

 that (the) user(S) can look around when moving toward his(THEIR) goal, the disadvantage is that he doesn't (THEY DON'T) have direct